### DISCUSSION TITLE
Dynamic head content not being rendered on production (Aure)

### DISCUSSION THREAD

**Discussion by Justincale** [oqtane/oqtane.framework#4231] [Q&A]
Hey guys,

I have a component that dynamically adds opengraph meta tags to the header via the AddHeadContent method. The Oqtane site is using Interactive Server Prerendering but this component overrides the RenderMode so that it renders Static:

public override string RenderMode => RenderModes.Static;

On my development machine, this works fine so that the opengraph tags get rendered to the head and are visible when I view the page source in Chrome.

The problem is, when deploying to my production server on Azure the opengraph tags (dynamic head content) are missing. Any ideas why this would be the case? 

The only difference i can see is that my local clr version is 8.0.4 and on Azure it is 8.0.1

Thanks

Here is a link if you wish to check it: https://ssdp.azurewebsites.net/news/!/Test-Article-1/1

**Comment by sbwalker** [comment]
@Justincale if your Site is using Interactive render mode it means that all components in the entire site will be rendering interactively. Blazor does not support a transition back to Static render mode if a parent component is Interactive. So even if you specify RenderMode => RenderModes.Static the component will be rendered interactively in your case.

The difference in CLR versions is most likely causing the problems you are seeing. Blazor had a LOT of issues in the initial .NET 8 release and the patch releases have contained a lot of critical fixes. You need to upgrade your Azure App Service to the latest SDK. This post provides details:

https://github.com/oqtane/oqtane.framework/discussions/4133

**Comment by Justincale** [comment]
Thanks Shaun, it was indeed the CLR version

**Reply by sbwalker** [reply]
I am glad you were able to resolve it. In total I probably spent an entire month of development effort chasing mysterious behaviors in Blazor in .NET 8... only to find out they were bugs which have now been fixed in the recent .NET SDKs.

**Reply by Justincale** [reply]
I don't envy you. The whole static rendering thing must be a headache too.
